chef/cookbooks/provisioner/libraries/repositories.rb

```diff
@@ -156,73 +72,16 @@ def inspect_repos(node)
       # This returns a hash containing the data about the repos that must be
       # used on nodes; optional repos (such as HA) will only be returned if
       # they can be used.
-      def get_repos(provisioner_server_node, platform, version)
-        admin_ip = Chef::Recipe::Barclamp::Inventory.get_network_by_type(provisioner_server_node, "admin").address
-        web_port = provisioner_server_node[:provisioner][:web_port]
-        provisioner_web = "http://#{admin_ip}:#{web_port}"
-        default_repos_url = "#{provisioner_web}/suse-#{version}/repos"
-
-        repos = Mash.new
-
-        case platform
-        when "suse"
-          repos = Mash.new
-          repos_from_attrs = suse_get_repos_from_attributes(provisioner_server_node,platform,version)
-
-          case version
-          when "11.3"
-            repo_names = %w(
-              SLES11-SP3-Pool
-              SLES11-SP3-Updates
-            )
-          when "11.4"
-            repo_names = %w(
-              SLES11-SP4-Pool
-              SLES11-SP4-Updates
-            )
-          when "12.0"
-            repo_names = %w(
-              SLES12-Pool
-              SLES12-Updates
-            )
-          when "12.1"
-            repo_names = %w(
-              SLES12-SP1-Pool
-              SLES12-SP1-Updates
-            )
-          else
-            raise "Unsupported version of SLE/openSUSE!"
-          end
-
-          # Add the new (not predefined) repositories from attributes
-          repos_from_attrs.each do |name,repo|
-            repo_names << name unless repo_names.include? name
-          end
-
-          # This needs to be done here rather than via deep-merge with static
-          # JSON due to the dynamic nature of the default value.
-          repo_names.each do |name|
-            repos[name] = repos_from_attrs.fetch(name, Mash.new)
-            repos[name][:url] ||= default_repos_url + "/" + name
-          end
-
-          # optional repos
-          unless provisioner_server_node[:provisioner][:suse].nil?
-            [[:common, :common_available],
-             [:cloud, :cloud_available],
-             [:hae, :hae_available],
-             [:storage, :storage_available]].each do |optionalrepo|
-              if provisioner_server_node[:provisioner][:suse][optionalrepo[1]]
-                suse_optional_repos(version, optionalrepo[0]).each do |name|
-                  repos[name] = repos_from_attrs.fetch(name, Mash.new)
-                  repos[name][:url] ||= default_repos_url + "/" + name
-                end
-              end
-            end
+      def get_repos(platform, version)
+        repositories = {}
+        repos_from_databag = Chef::DataBag.load("repositories").keys rescue []
```

Comment:
Style/RescueModifier: Avoid using `rescue` in its modifier form. (https://github.com/bbatsov/ruby-style-guide#no-rescue-modifiers)